src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -169,7 +172,8 @@ protected AbstractBatch(
         final long maxAwaitTimeShutdown,
         @Nullable final BatchListener batchListener,
         final int maxFlushRetries,
-        final long flushRetryDelay) {
+        final long flushRetryDelay,
+        @Nullable final Logger logger) {
```

Comment:
Shouldn't this be a logger just for confidential logging purposes? We maintain the old logger to log non-confidential data and use the log that is passed here as an argument to log just confidential data... If this is received as null we log everything with the other logger that we had in the class before.

Reply:
The idea here is to allow a custom `Logger` to be sent as parameter without introducing any breaking changes to the current behavior PDB already has. Confidential logger is not a concept from PDB, it is from Pulse. That is why the code uses a generic Logger instance. After these changes are applied to PDB, Pulse will need to be changed as well, so it can call the correct `createBatch()`, which receives a confidential logger.